File: app/src/main/java/com/sami/rippel/livewallpapers/lwpskyview/WallpaperRenderer.java
package com.sami.rippel.livewallpapers.lwpskyview;

import android.content.Context;
import android.view.MotionEvent;

import com.sami.rippel.allah.R;

import org.rajawali3d.math.vector.Vector3;
import org.rajawali3d.renderer.Renderer;

public class WallpaperRenderer extends Renderer {

    private String axe = "";
    private float downX;
    private float downY;

    public WallpaperRenderer(Context context) {
        super(context);
    }

    @Override
    public void onOffsetsChanged(float v, float v2, float v3, float v4, int i, int i2) {
    }

    @Override
    public void onTouchEvent(MotionEvent event) {
        switch (event.getAction()) {
            case MotionEvent.ACTION_DOWN: {
                downX = event.getX();
                downY = event.getY();
            }
            case MotionEvent.ACTION_UP: {
                float upX = event.getX();
                float upY = event.getY();
                float deltaX = downX - upX;
                float deltaY = downY - upY;
                int min_distance = 100;
                if (Math.abs(deltaX) > Math.abs(deltaY)) {
                    if (Math.abs(deltaX) > min_distance) {
                        if (deltaX < 0) {
                            axe = "LefttoRigh";
                        }
                        if (deltaX > 0) {
                            axe = "RighToLeft";
                        }
                    }
                }
                else {
                    if (Math.abs(deltaY) > min_distance) {
                        if (deltaY < 0) {
                            axe = "TopToBottom";
                        }
                        if (deltaY > 0) {
                            axe = "BottomToTop";
                        }
                    }
                }
            }
        }
    }


    @Override
    protected void onRender(long ellapsedRealtime, double deltaTime) {
        super.onRender(ellapsedRealtime, deltaTime);
        try {
            switch (axe) {
                case "LefttoRigh":
                    getCurrentCamera().rotate(Vector3.Axis.Y, -0.1);
                    break;
                case "RighToLeft":
                    getCurrentCamera().rotate(Vector3.Axis.Y, +0.1);
                    break;
                case "TopToBottom":
                    getCurrentCamera().rotate(Vector3.Axis.X, -0.1);
                    break;
                case "BottomToTop":
                    getCurrentCamera().rotate(Vector3.Axis.X, +0.1);
                    break;
            }
        } catch (Exception ignored) {
        }
    }

    @Override
    protected void initScene() {
        try {
            getCurrentScene().setSkybox(R.drawable.night);
        } catch (Exception ignored) {
        }
    }
}

File: ui-sky-rajawali-lwp/src/main/java/com/sfaxdroid/sky/WallpaperRenderer.kt
package com.sfaxdroid.sky

import android.content.Context
import android.view.MotionEvent
import org.rajawali3d.math.vector.Vector3
import org.rajawali3d.renderer.Renderer
import kotlin.math.abs

class WallpaperRenderer(context: Context) : Renderer(context) {

    private var axe: Position = Position.None
    private var downX = 0f
    private var downY = 0f

    override fun onOffsetsChanged(
        v: Float,
        v2: Float,
        v3: Float,
        v4: Float,
        i: Int,
        i2: Int
    ) {
    }

    override fun onTouchEvent(event: MotionEvent) {
        when (event.action) {
            MotionEvent.ACTION_DOWN -> {
                run {
                    downX = event.x
                    downY = event.y
                }
                run {
                    val upX = event.x
                    val upY = event.y
                    val deltaX = downX - upX
                    val deltaY = downY - upY
                    val minDistance = 100
                    if (abs(deltaX) > abs(deltaY)) {
                        if (abs(deltaX) > minDistance) {
                            if (deltaX < 0) {
                                axe = Position.LeftToRights
                            }
                            if (deltaX > 0) {
                                axe = Position.RightToLeft
                            }
                        }
                    } else {
                        if (abs(deltaY) > minDistance) {
                            if (deltaY < 0) {
                                axe = Position.TopToBottom
                            }
                            if (deltaY > 0) {
                                axe = Position.BottomToTop
                            }
                        }
                    }
                }
            }
            MotionEvent.ACTION_UP -> {
                val upX = event.x
                val upY = event.y
                val deltaX = downX - upX
                val deltaY = downY - upY
                val minDistance = 100
                if (abs(deltaX) > abs(deltaY)) {
                    if (abs(deltaX) > minDistance) {
                        if (deltaX < 0) {
                            axe = Position.LeftToRights
                        }
                        if (deltaX > 0) {
                            axe = Position.RightToLeft
                        }
                    }
                } else {
                    if (abs(deltaY) > minDistance) {
                        if (deltaY < 0) {
                            axe = Position.TopToBottom
                        }
                        if (deltaY > 0) {
                            axe = Position.BottomToTop
                        }
                    }
                }
            }
        }
    }

    override fun onRender(
        ellapsedRealtime: Long,
        deltaTime: Double
    ) {
        super.onRender(ellapsedRealtime, deltaTime)
        rotateCamera()
    }

    private fun rotateCamera() {
        when (axe) {
            is Position.LeftToRights -> currentCamera.rotate(Vector3.Axis.Y, -0.1)
            is Position.RightToLeft -> currentCamera.rotate(Vector3.Axis.Y, +0.1)
            is Position.TopToBottom -> currentCamera.rotate(Vector3.Axis.X, -0.1)
            is Position.BottomToTop -> currentCamera.rotate(Vector3.Axis.X, +0.1)
        }
    }

    override fun initScene() {
        try {
            currentScene.setSkybox(R.drawable.night)
        } catch (ignored: Exception) {
        }
    }

    sealed class Position {
        object LeftToRights : Position()
        object RightToLeft : Position()
        object TopToBottom : Position()
        object BottomToTop : Position()
        object None : Position()
    }

}